locator/everywhere_replication_strategy.hh

```diff
@@ -20,7 +20,9 @@ public:
 
     virtual future<host_id_set> calculate_natural_endpoints(const token& search_token, const token_metadata& tm) const override;
 
-    virtual void validate_options(const gms::feature_service&, const locator::topology&) const override { /* noop */ }
+    virtual void validate_options(const gms::feature_service& fs, const locator::topology& t) const override {
+        abstract_replication_strategy::validate_options(fs, t);
```

Comment:
If it just calls the base-class validate_options, why not just remove this method and let it inherit it natively?

Reply:
IDK what's the best approach here. The base class function is marked as pure virtual:
```
    virtual void validate_options(const gms::feature_service&) const = 0;
```
in `abstract_replication_strategy.hh`, so I just provided an implementation to it in the corresponding source file, so that child classes can call it. This way I left it pure virtual still and allowed to share the code. Maybe I could remove the ` = 0`, but I don't understand why was it introduced in the first place.

Reply:
Ah, I see. It's indeed that validate_options() is pure virtual in the base class in master. Then this change is OK